models/HubSpotHttpError.ts

```diff
@@ -1,41 +1,239 @@
+        options.cause.name !== 'Error' ? [`${options.cause.name}:`] : [];
+
+      if (options.cause.message) {
+        messages.push(options.cause.message);
+      }
+
+      if ('reason' in options.cause && options.cause.reason) {
+        messages.push(`${options.cause.reason}`);
+      }
+
+      this.message = messages.join(' ');
+    }
+  }
+
+  public updateContext(context: Partial<HubSpotHttpErrorContext>) {
+    this.context = { ...this.context, ...context };
+  }
+
+  public toString() {
+    let baseString = `${this.name}: \n- message: ${this.message}`;
+    if (this.validationErrors && this.validationErrors.length > 0) {
+      baseString = `${baseString} \n- errors: ${this.validationErrors.join('\n- ')}`;
+    }
+    if (this.context) {
+      baseString = `${baseString} \n- context: ${JSON.stringify(this.context, undefined, 2)}`;
+    }
+    return baseString;
+  }
+
+  private updateContextFromCause(
+    cause: AxiosError,
+    context?: HubSpotHttpErrorContext
+  ) {
+    const generatedContext: HubSpotHttpErrorContext = {};
+    if (!cause) {
+      return;
+    }
+
+    try {
+      generatedContext.accountId = cause.config?.params?.portalId;
+      generatedContext.payload = JSON.stringify(cause.config?.data);
+      // This will just be the url path
+      generatedContext.request = cause.config?.url;
+    } catch (e) {
+      logger.debug(e);
+    }
+
+    // Allow the provided context to override the generated context
+    this.context = { ...generatedContext, ...context };
+  }
+
+  private parseValidationErrors(
+    responseData: {
+      errors?: Array<ValidationError>;
+      message?: string;
+    } = { errors: [], message: '' }
+  ): void {
+    if (!responseData) {
+      return;
+    }
+
+    const errorMessages = [];
+
+    const { errors, message } = responseData;
+
+    if (message) {
+      errorMessages.push(message);
+    }
+
+    if (errors) {
+      const specificErrors = errors.map(error => {
+        let errorMessage = error.message;
+        if (error.errorTokens && error.errorTokens.line) {
+          errorMessage = `line ${error.errorTokens.line}: ${errorMessage}`;
+        }
+        return errorMessage;
+      });
+      errorMessages.push(...specificErrors);
+    }
+
+    this.validationErrors = errorMessages;
+  }
+
+  private joinErrorMessages(
+    error: AxiosError<{ message: string; errors: { message: string }[] }>,
+    context: HubSpotHttpErrorContext = {}
+  ): void {
+    const i18nKey = 'errors.apiErrors';
+    const status = error.response?.status;
+    const method = error.config?.method as HttpMethod;
+    const { projectName } = context;
+
+    let messageDetail: string;
+
+    if (context.accountId) {
+      const action =
+        (method && HTTP_METHOD_VERBS[method]) || HTTP_METHOD_VERBS.get;
+
+      const preposition =
+        (method && HTTP_METHOD_PREPOSITIONS[method]) ||
+        HTTP_METHOD_PREPOSITIONS.get;
+
+      const requestName = context.request
```

Comment:
Funny enough, I think you actually fixed this logic. I'm not sure which version I like better. Showing them the exact endpoint that we failed to hit seems like it might be more information than the end user actually needs. But maybe this is a case where the CLI should be intercepting this and logging something more user-friendly to the end user?

<img width="804" alt="image" src="https://github.com/user-attachments/assets/6a77a072-83ed-46f4-86d9-e2ca540e07bd">

Reply:
I think if we think about in in the context of the local-dev-lib end user vs the CLI end user it could make sense to do it this way

Reply:
I agree with Camden that from the library perspetive it makes sense to give as informative of error messages as possible.  I think the problem actually came do to me trying to get clever and auto generate the context object from the `AxiosError`, which always gives us maximum detail. I think I found a good middle ground of having `message` in the `HubSpotHttpError` be a high level message that only has the `accountId` for the context.  And then having `internalMessage` (although maybe `detailedMessage` is a better name), be the message with the full context.  The `internalMessage` is what gets logged out when we call the `toString` method.

This gives us the flexibility to add the context when wanted to user facing errors, while making sure that we still capture the full error for debugging.